src/axelar/AxelarTransceiver.sol

```diff
@@ -0,0 +1,182 @@
+    /// @param deliveryPayment the amount of native tokens to be used as gas for delivery.
+    /// @param recipientNttManagerAddress the address of the NttManager to receive this message.
+    /// @param refundAddress the address to receive the gas refund if overpayed.
+    /// @param nttManagerMessage A message to be sent to the nttManager on the recipient chain.
+    function _sendMessage(
+        uint16 recipientChainId,
+        uint256 deliveryPayment,
+        address, /*caller*/
+        bytes32 recipientNttManagerAddress,
+        bytes32 refundAddress,
+        TransceiverStructs.TransceiverInstruction memory, /*transceiverInstruction*/
+        bytes memory nttManagerMessage
+    ) internal virtual override onlyNttManager {
+        AxelarTransceiverStorage storage slot = _storage();
+        string memory destinationContract = slot.idToTransceiverAddress[recipientChainId];
+        string memory destinationChain = slot.idToAxelarChainId[recipientChainId];
+
+        if (bytes(destinationChain).length == 0 || bytes(destinationContract).length == 0) {
+            revert InvalidChainId(recipientChainId);
+        }
+
+        bytes memory payload = abi.encode(nttManager, nttManagerMessage, recipientNttManagerAddress);
+
+        _callContract(
+            destinationChain,
+            destinationContract,
+            payload,
+            fromWormholeFormat(refundAddress),
+            deliveryPayment
+        );
+
+        emit SendTransceiverMessage(
+            recipientChainId, nttManagerMessage, recipientNttManagerAddress, refundAddress
+        );
+    }
+
+    function _callContract(
+        string memory destinationChain,
+        string memory destinationContract,
+        bytes memory payload,
+        address refundAddress,
+        uint256 deliveryPayment
+    ) internal virtual {
+        gasService.payGas{value: deliveryPayment}(
+            address(this),
+            destinationChain,
+            destinationContract,
+            payload,
+            0,
+            false,
+            refundAddress,
+            bytes("")
+        );
+
+        gateway().callContract(destinationChain, destinationContract, payload);
+    }
+
+    function _execute(
+        bytes32, /*commandId*/
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) internal override {
+        AxelarTransceiverStorage storage slot = _storage();
+        uint16 sourceChainId = slot.axelarChainIdToId[sourceChain];
+        if (sourceChainId == 0 || slot.transceiverAddressToId[sourceAddress] != sourceChainId) {
+            revert InvalidSibling(sourceChainId, sourceChain, sourceAddress);
+        }
+
+        (
+            bytes32 sourceNttManagerAddress,
+            TransceiverStructs.NttManagerMessage memory nttManagerMessage,
+            bytes32 recipientNttManagerAddress
+        ) = abi.decode(payload, (bytes32, TransceiverStructs.NttManagerMessage, bytes32));
```

Comment:
This decoding is incorrect and will result in a malformed manager message. The payload length of the payload in a manager message is encoded as a uint16, but the implicit nested `abi.decode` of the struct will read 32 bytes instead.

Suggestion is to decode the payload into `(bytes32, bytes, bytes32)` and then call `parseNttManagerMessage` in `TransceiverStructs.sol` to parse the byte array safely into the manager message.

We also suggest adding some tests for the execution side.

Reply:
fixed